dags/utils/dataproc.py

```diff
@@ -0,0 +1,457 @@
+                                        the environment variables for the new process 
+                                        (templated).
+    :param str arguments:               Passed to `airflow_gcp.sh`, passed as one long string 
+                                        of space separated args.
+    :param str artifact_bucket:         The bucket with script-runner.jar and airflow_gcp.sh.
+
+    """
+
+    if job_name is None or uri is None or cluster_name is None:
+        raise AirflowException('Please specify job_name, uri, and cluster_name.')
+
+    dataproc_helper = DataProcHelper(cluster_name=cluster_name,
+                                     num_workers=num_workers,
+                                     image_version=image_version,
+                                     zone=zone,
+                                     idle_delete_ttl=idle_delete_ttl,
+                                     auto_delete_ttl=auto_delete_ttl,
+                                     master_machine_type=master_machine_type,
+                                     worker_machine_type=worker_machine_type,
+                                     num_preemptible_workers=num_preemptible_workers,
+                                     service_account=service_account,
+                                     artifact_bucket=artifact_bucket,
+                                     aws_conn_id=aws_conn_id,
+                                     gcp_conn_id=gcp_conn_id)
+
+    _dag_name = '{}.{}'.format(parent_dag_name, dag_name)
+    environment = _format_envvar(env)
+    jar_url = 'gs://{}/bin/script-runner.jar'.format(artifact_bucket)
+
+    args = [
+        'gs://{}/bootstrap/airflow_gcp.sh'.format(artifact_bucket),
+        '--job-name', job_name,
+        '--uri', uri,
+        '--environment', environment
+    ]
+
+    if arguments:
+        args += ['--arguments', arguments]
+
+    with models.DAG(_dag_name, default_args=default_args) as dag:
+        create_dataproc_cluster = dataproc_helper.create_cluster()
+
+        # Run DataprocSparkOperator with script-runner.jar pointing to airflow_gcp.sh.
+        # Note - When we upgrade to a later version of Airflow that pulls in latest
+        # DataProcSparkOperator code, use the argument main_jar=jar_url instead, and
+        # remove arguments main_class and dataproc_spark_jars.
+        run_script_on_dataproc = DataProcSparkOperator(
+            cluster_name=cluster_name,
+            task_id='run_script_on_dataproc',
+            job_name=job_name,
+            dataproc_spark_jars=[jar_url],
+            main_class='com.amazon.elasticmapreduce.scriptrunner.ScriptRunner',
+            arguments=args,
+            gcp_conn_id=gcp_conn_id)
+
+        delete_dataproc_cluster = dataproc_helper.delete_cluster()
+
+        create_dataproc_cluster >> run_script_on_dataproc >> delete_dataproc_cluster
+        return dag
+# End moz_dataproc_scriptrunner
+
+
+class DataProcHelper():
```

Comment:
```suggestion
class DataProcHelper:
```

Reply:
It might be less confusing to put the definition before its usage.

Reply:
done